fabric-network/lib/contract.js

```diff
@@ -197,6 +198,48 @@ class Contract {
 		return listener;
 	}
 
+	/**
+	 * Provide a Discovery Interest settings to help the peer's discovery service
+	 * build an endorsement plan. This chaincode Id will be include by default in
+	 * the list of discovery interests. If this contract's chaincode is in one or
+	 * more collections then use this method with this chaincode Id to change the
+	 * default discovery interest to include those collection names.
+	 * @param {DiscoveryIntereset} interest - These will be added to the
+	 * existing discovery interests to be used as the "endorsement_hint"
+	 * attribute of the [request]{@link ChaincodeInvokeRequest}
+	 * when the {@link module:fabric-network.Transaction#submit}
+	 * performs a {@link Channel#sendTransactionProposal}
+	 */
+	addDiscoveryInterest(interest) {
+		if (!(typeof interest === 'object')) {
+			throw Error('"interest" parameter must be a DiscoveryInterest object');
+		}
+
+		let found = false;
```

Comment:
Logically correct but could break once it finds a matching existing entry. How about something like this for clarity?
```javascript
const existingIndex = this.discoveryInterests.findIndex((entry) => entry.name === interest.name);
if (existingIndex >= 0) {
    this.discoveryInterests[existingIndex] = interest;
} else {
    this.discoveryInterests.push(interest);
}
```

Reply:
fixed